src/main/java/bio/terra/metadata/Study.java

```diff
@@ -1,72 +1,80 @@
-        studySpecification.getTables().forEach(tableModel ->
-            tables.put(tableModel.getName(), new StudyTable(tableModel)));
-        studySpecification.getRelationships().forEach(relationship ->
-            relationships.put(relationship.getName(), new StudyRelationship(relationship, tables)));
-        studySpecification.getAssets().forEach(asset ->
-            assetSpecifications.put(asset.getName(), new AssetSpecification(asset, tables, relationships)));
+    public String getName() {
+        return name;
     }
+    public Study setName(String name) { this.name = name; return this; }
 
-    public Study(String name, String description) {
-        this.name = name;
-        this.description = description;
+    public String getDescription() {
+        return description;
     }
+    public Study setDescription(String description) { this.description = description; return this; }
 
-    // Constructor for building studies in unit tests.
-    public Study(String name, String description, Map<String, StudyTable> tables) {
-        this(name, description);
-        this.tables = tables;
+    public Instant getCreatedDate() {
+        return createdDate;
     }
+    public Study setCreatedDate(Instant createdDate) { this.createdDate = createdDate; return this; }
 
-    public StudySummaryModel toSummary() {
-        return new StudySummaryModel()
-                .id(this.id.toString())
-                .name(this.name)
-                .description(this.description);
+    public List<StudyTable> getTables() {
+        return tables;
     }
-
-    public UUID getId() {
-        return id;
+    public Study setTables(List<StudyTable> tables) {
+        this.tables = Collections.unmodifiableList(tables);
+        return this;
     }
 
-    public void setId(UUID id) { this.id = id; }
-
-    public String getName() {
-        return name;
+    public List<StudyRelationship> getRelationships() {
+        return relationships;
+    }
+    public Study setRelationships(List<StudyRelationship> relationships) {
+        this.relationships = Collections.unmodifiableList(relationships);
+        return this;
     }
 
-    public String getDescription() {
-        return description;
+    public List<AssetSpecification> getAssetSpecifications() {
+        return assetSpecifications;
+    }
+    public Study setAssetSpecifications(List<AssetSpecification> assetSpecifications) {
+        this.assetSpecifications = Collections.unmodifiableList(assetSpecifications);
+        return this;
     }
 
-    public Collection<StudyTable> getTables() {
-        return Collections.unmodifiableCollection(tables.values());
+    public Map<UUID, StudyTableColumn> getAllColumnsById() {
+        Map<UUID, StudyTableColumn> columns = new HashMap<>();
+        getTables().forEach(table -> table.getColumns().forEach(column -> columns.put(column.getId(), column)));
+        return columns;
```

Comment:
I notice you removed the `unmodifiable` from this and the methods below.
Was that on purpose?

Reply:
yes. before the map was the data member and i didn't want someone adding to it after i returned it. now, when these methods are called i'm constructing a map and then returning it. i don't care if someone wants to modify it - it doesn't affect this object's internal state.